src/test/java/de/tum/cit/aet/artemis/programming/ProgrammingExerciseIntegrationTestService.java

```diff
@@ -1653,26 +1622,22 @@ void updateTestCases_testCaseBonusPointsNull() throws Exception {
         mockDelegate.mockTriggerBuild(programmingExercise.getTemplateParticipation());
 
         final var updates = transformTestCasesToDto(testCases);
-        updates.getFirst().setBonusPoints(null);
+        var firstUpdate = updates.getFirst();
+        firstUpdate = new ProgrammingExerciseTestCaseDTO(firstUpdate.id(), firstUpdate.weight(), firstUpdate.bonusMultiplier(), null, firstUpdate.visibility());
+        updates.set(0, firstUpdate);
+
         final var endpoint = "/programming-exercises/" + programmingExercise.getId() + "/update-test-cases";
 
         final var testCasesResponse = request.patchWithResponseBody("/api" + endpoint, updates, new TypeReference<List<ProgrammingExerciseTestCase>>() {
         }, HttpStatus.OK);
-        final var updatedTestCase = testCasesResponse.stream().filter(testCase -> testCase.getId().equals(updates.getFirst().getId())).findFirst().orElseThrow();
+        final var updatedTestCase = testCasesResponse.stream().filter(testCase -> testCase.getId().equals(updates.getFirst().id())).findFirst().orElseThrow();
         assertThat(updatedTestCase.getBonusPoints()).isZero();
         assertThat(testCasesResponse.stream().filter(testCase -> !testCase.getId().equals(updatedTestCase.getId()))).allMatch(testCase -> testCase.getBonusPoints() == 1d);
     }
 
     private static List<ProgrammingExerciseTestCaseDTO> transformTestCasesToDto(Collection<ProgrammingExerciseTestCase> testCases) {
-        return testCases.stream().map(testCase -> {
-            final var testCaseUpdate = new ProgrammingExerciseTestCaseDTO();
-            testCaseUpdate.setId(testCase.getId());
-            testCaseUpdate.setVisibility(testCase.getVisibility());
-            testCaseUpdate.setWeight(testCase.getWeight());
-            testCaseUpdate.setBonusMultiplier(testCase.getBonusMultiplier());
-            testCaseUpdate.setBonusPoints(testCase.getBonusPoints());
-            return testCaseUpdate;
-        }).toList();
+        return testCases.stream().map(testCase -> new ProgrammingExerciseTestCaseDTO(testCase.getId(), testCase.getWeight(), testCase.getBonusMultiplier(),
+                testCase.getBonusPoints(), testCase.getVisibility())).toList();
```

Comment:
Since you want to modify the returned list later on, you cannot use `toList`, since it returns an unmodifiable list.
```suggestion
        return testCases.stream().map(testCase -> new ProgrammingExerciseTestCaseDTO(testCase.getId(), testCase.getWeight(), testCase.getBonusMultiplier(),
                testCase.getBonusPoints(), testCase.getVisibility())).collect(Collectors.toCollection(ArrayList::new));
```